### ISSUE TITLE
Question : How I can check whether candle is closed just above /below MA line and pattern detection

### ISSUE THREAD

**Issue by yogikorke** [ta4j/ta4j#246]
I was wondering how to check candle closed, opened just above or below some MA line. Also how to detect patterns like double bottom from timeseries? Any ideas appreciated. Thanks in advance.

**Comment by edlins** [comment]
A simple example comparing close prices to its SMA:
```
        ClosePriceIndicator closePrice = new ClosePriceIndicator(series);
        SMAIndicator sma = new SMAIndicator(closePrice, 14);
        Rule entryRule = new OverIndicatorRule(closePrice, sma);
        Rule exitRule = new UnderIndicatorRule(closePrice, sma);
        Strategy strategy = new BaseStrategy(entryRule, exitRule);
        TimeSeriesManager seriesManager = new TimeSeriesManager(series);
        TradingRecord tradingRecord = seriesManager.run(strategy);
```
As far as detecting highs/tops and lows/bottoms, I'd love to see `boolean HighsIndicator` and `LowsIndicator`.  There's an algorithm [here](https://www.marketsmadeclear.com/Resources/Currency-Strength-Matrix/How-to-read-a-chart.aspx) but I don't think ta4j has anything like that yet.  Those would be great contributions.

**Comment by edlins** [comment]
> there already is a IsHighestRule and a IsLowestRule

Yes, but the algo I referenced is much more involved.  One could probably use `IsHighestRule` to help implement `IsTopRule` but `IsHighestRule` implements only one of the six rules documented at marketsmadeclear.com.
